src/transaction_signature_options/sign_with_access_key_file/mod.rs

```diff
@@ -89,9 +133,109 @@ impl From<SignAccessKeyFileContext> for super::SubmitContext {
+                Err(err) => return interactive_clap::ResultFromCli::Err(Some(clap_variant), err),
+            };
+        }
+        let nonce = clap_variant.nonce;
+        if clap_variant.block_hash.is_none() {
+            clap_variant.block_hash = match Self::input_block_hash(&context) {
+                Ok(optional_block_hash) => optional_block_hash,
+                Err(err) => return interactive_clap::ResultFromCli::Err(Some(clap_variant), err),
+            };
+        }
+        let block_hash = clap_variant.block_hash.clone();
+        if clap_variant.meta_transaction_valid_for.is_none() {
+            clap_variant.meta_transaction_valid_for =
+                match Self::input_meta_transaction_valid_for(&context) {
+                    Ok(meta_transaction_valid_for) => meta_transaction_valid_for,
+                    Err(err) => {
+                        return interactive_clap::ResultFromCli::Err(Some(clap_variant), err)
+                    }
+                };
+        }
+        let meta_transaction_valid_for = clap_variant.meta_transaction_valid_for;
+
+        let new_context_scope = InteractiveClapContextScopeForSignAccessKeyFile {
+            file_path,
+            nonce,
+            block_hash,
+            meta_transaction_valid_for,
+        };
+        let output_context =
+            match SignAccessKeyFileContext::from_previous_context(context, &new_context_scope) {
+                Ok(new_context) => new_context,
+                Err(err) => return interactive_clap::ResultFromCli::Err(Some(clap_variant), err),
+            };
+
+        match super::Submit::from_cli(clap_variant.submit.take(), output_context.into()) {
+            interactive_clap::ResultFromCli::Ok(submit) => {
+                clap_variant.submit = Some(submit);
+                interactive_clap::ResultFromCli::Ok(clap_variant)
+            }
+            interactive_clap::ResultFromCli::Cancel(optional_submit) => {
+                clap_variant.submit = optional_submit;
+                interactive_clap::ResultFromCli::Cancel(Some(clap_variant))
+            }
+            interactive_clap::ResultFromCli::Back => interactive_clap::ResultFromCli::Back,
+            interactive_clap::ResultFromCli::Err(optional_submit, err) => {
+                clap_variant.submit = optional_submit;
+                interactive_clap::ResultFromCli::Err(Some(clap_variant), err)
+            }
+        }
+    }
+}
+
+impl SignAccessKeyFile {
+    fn input_nonce(
+        _context: &crate::commands::TransactionContext,
+    ) -> color_eyre::eyre::Result<Option<u64>> {
+        Ok(None)
+    }
+
+    fn input_block_hash(
+        _context: &crate::commands::TransactionContext,
+    ) -> color_eyre::eyre::Result<Option<String>> {
+        Ok(None)
+    }
+
+    fn input_meta_transaction_valid_for(
+        _context: &crate::commands::TransactionContext,
+    ) -> color_eyre::eyre::Result<Option<u64>> {
+        Ok(Some(1000))
```

Comment:
As we discussed before, it should be `None` as otherwise `to_cli_args` will print `--meta-transaction-valid-for 1000` when a user did not even specify it.

```suggestion
        Ok(None)
```